### DISCUSSION TITLE
Where is the setting to turn off Vsync?

### DISCUSSION THREAD

**Discussion by tictac1234** [AvaloniaUI/Avalonia#14656] [Q&A]
Hello.
In AvaloniaUI, Where is the setting to turn off Vsync?

I'm sorry if this topic has already been answered...

**Comment by maxkatz6** [comment]
There is no direct control over your render timer/vsync. 
On most platforms it is enabled. And only on Windows you can disable it by not using Win32CompositionMode.WinUIComposition or DirectComposition. In BuildAvaloniaApp:
```
.With(new Win32PlatformOptions
{
    CompositionMode = new [] { Win32CompositionMode.RedirectionSurface } 
})
```

This change also will remove ability to have window effects like transparency and blur. Also, not sure if it will even support hardware acceleration this way.

Why do you need to disable VSync?

**Reply by tictac1234** [reply]
Thank you.

I created a test project in WPF(d3dimage), SharpDX and tried the mouse movement detection.
I felt it was a little slow, so I turned off Vsync as shown below and the detection was much faster.

`mSwapChain.Present( 0, PresentFlags.None);`

When I set syncinterval to 0, the mouse detection delay was almost gone, so I wanted to know if I could set this in AvaloniaUI as well.

However, I wrote the code you gave me, but the mouse detection delay did not change.

Is it possible to create a custom view in AvaloniaUI, incorporate DirectX11 or Metal, and turn off Vsync?

(In case you are wondering, I have mentioned this in an earlier topic)

https://github.com/AvaloniaUI/Avalonia/discussions/14062

**Reply by Butcher-vj** [reply]
> There is no direct control over your render timer/vsync. On most platforms it is enabled. And only on Windows you can disable it by not using Win32CompositionMode.WinUIComposition or DirectComposition. In BuildAvaloniaApp:
> 
> ```
> .With(new Win32PlatformOptions
> {
>     CompositionMode = new [] { Win32CompositionMode.RedirectionSurface } 
> })
> ```
> 
> This change also will remove ability to have window effects like transparency and blur. Also, not sure if it will even support hardware acceleration this way.
> 
> Why do you need to disable VSync?
Hello, @maxkatz6 ,
I want to achieve acrylic blur effect like this,
TransparencyLevelHint="AcrylicBlur"
Background="Transparent"
But it is flickering while recording the screen, so i found something like this to add win32platformoptions.compostionmode 
 .With(new Win32PlatformOptions
 {
     ShouldRenderOnUIThread = true,
     CompositionMode = new[]
     {
         Win32CompositionMode.WinUIComposition
     },
     RenderingMode = new[]
     {
         Win32RenderingMode.AngleEgl
     }
 })
 But even though i did like this, still my app is flickering while screen recording or screen sharing.
 OS - Windows 11
Avalonia - 11.0.11
